docs/weblog/README.md

```diff
@@ -0,0 +1,99 @@
+# Weblog
+
+A weblog is a web app that system uses to test the library. A weblog app is required for each platform that the system tests will test. The weblog must implement a number of different end points.
+
+## Disclaimer
+
+This document describes endpoints implemented on weblog. Though, it's not a complete description, and can contains mistakes. The source of truth are the test itself. If a weblog endpoint passes system tests, then you can consider it as ok. And if it does not passes it, then you must correct it, even if it's in line with this document.
+
+**You are strongly encouraged to help others by submitting corrections when you notice issues with this document.**
+
+# Endpoints
+
+## GET / (aka plain text end point)
+
+The following text should be written to the body of the response:
+
+```
+Hello world!\n
+```
+
+## GET /headers
+
+This end point should set the following headers:
+
+```
+content-type: text
+content-length: 16
```

Comment:
watch out for setting `Content-Length` here: it can produce a wait til a timeout occurs as some parts of a framework or user agent may wait for data that never comes. It is often set by the framework anyway, the important part appears to be that it MUST be present, whether manually or via framework facilities.